src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -247,6 +249,52 @@ void ManageSurfaceHeatBalance(EnergyPlusData &state)
 // Beginning Initialization Section of the Module
 //******************************************************************************
 
+Real64 GetSurfAddOnOverrideValue(EnergyPlusData &state, int surfNum, Material::VariableAbsCtrlSignal controlSignal, int scheduleIdx, int functionIdx)
+{
+    Real64 overrideValue;
+    if (controlSignal == Material::VariableAbsCtrlSignal::Scheduled) {
+        overrideValue = ScheduleManager::GetCurrentScheduleValue(state, scheduleIdx);
+        return max(min(overrideValue, 0.9999), 0.0001);
+    } else {
+        Real64 triggerValue;
+        if (controlSignal == Material::VariableAbsCtrlSignal::SurfaceTemperature) {
+            triggerValue = state.dataHeatBalSurf->SurfTempOut(surfNum);
+            overrideValue = Curve::CurveValue(state, functionIdx, triggerValue);
+        } else if (controlSignal == Material::VariableAbsCtrlSignal::SurfaceReceivedSolarRadiation) {
+            triggerValue = state.dataHeatBal->SurfQRadSWOutIncident(surfNum);
+            overrideValue = Curve::CurveValue(state, functionIdx, triggerValue);
+        } else { // controlled by heating cooling mode
+            int zoneNum = state.dataSurface->Surface(surfNum).Zone;
+            bool isCooling = (state.dataZoneEnergyDemand->ZoneSysEnergyDemand(zoneNum).TotalOutputRequired < 0);
+            triggerValue = static_cast<Real64>(isCooling);
+            overrideValue = Curve::CurveValue(state, functionIdx, triggerValue);
+        }
+        return max(min(overrideValue, 0.9999), 0.0001);
+    }
+}
+
+void MaterialAddOnOverride(EnergyPlusData &state)
+{
+    for (int surfNum : state.dataSurface->AllVaryThermalAbsOpaqSurfaceList) {
```

Comment:
Is it possible for the same surface to appear in both lists? In that case, the override value from one would overwrite the override value from the other, no?